### PR TITLE
* FIX [mqtt_parser] fix https://github.com/emqx/nanomq/issues/1181 in…

### PR DESCRIPTION
**Pull request by JaylinYu**

### CODE REVIEW

File: src/sp/protocol/mqtt/mqtt_parser.c [hunk 1]
```diff
@@ -597,26 +604,30 @@ conn_handler(uint8_t *packet, conn_param *cparam, size_t max)
...
-	if (cparam->will_flag == 1 && cparam->will_qos > 2)
-		return PROTOCOL_ERROR;
-	pos++;
-	// keepalive
-	NNI_GET16(packet + pos, tmp);
-	cparam->keepalive_mqtt = tmp;
-	pos += 2;
-	// properties
+	rv                   = (len_of_str < 0 && pos + 4 < max) ? PROTOCOL_ERROR : 0;
+	if (rv == 0) {
```
Comment (wanghaEMQ) [inline src/sp/protocol/mqtt/mqtt_parser.c:608]: Try the codes follows.
```
if (rv != 0)
    return rv;
```
Then keep the rest part of the original codes.

Comment (JaylinYu) [reply src/sp/protocol/mqtt/mqtt_parser.c:608]: ok fine

File: src/sp/protocol/mqtt/mqtt_parser.c [hunk 2]
```diff
@@ -243,6 +245,11 @@ copyn_utf8_str(const uint8_t *src, uint32_t *pos, int *str_len, int limit)
 			memcpy(dest, src + (*pos), *str_len);
 			dest[*str_len] = '\0';
 			*pos           = (*pos) + (*str_len);
+			if (*pos >= max) {
```
Comment (Hermann0222) [inline src/sp/protocol/mqtt/mqtt_parser.c:248]: >= or >

Comment (JaylinYu) [reply src/sp/protocol/mqtt/mqtt_parser.c]: gotcha

**Review by Hermann0222 — APPROVED** [review]

**Pull request merged**